src/python_testing/TC_IDM_1_4.py

```diff
@@ -247,8 +247,65 @@ async def steps_3_to_9(self, dummy_value):
+        asserts.assert_true(has_data_model_test_feature, "DataModelTest Feature is not supported by DUT")
+
+        self.step(10)
+        try:
+            test_event_triggers_enabled = await self.read_single_attribute(
+                dev_ctrl,
+                dut_node_id,
+                endpoint=0,
+                attribute=Clusters.GeneralDiagnostics.Attributes.TestEventTriggersEnabled
+            )
+        except InteractionModelError:
+            asserts.fail("DUT failed to respond reading TestEventTriggersEnabled attribute")
+        asserts.assert_true(test_event_triggers_enabled, "Test Event Triggers are NOT Enabled on DUT")
+
+        self.step(11)
+        asserts.assert_true('PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY' in self.matter_test_config.global_test_params,
+                            "PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY must be included on the command line in "
+                            "the --hex-arg flag as PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY:<key>, "
+                            "e.g. --hex-arg PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY:000102030405060708090a0b0c0d0e0f")
+        enable_key = self.matter_test_config.global_test_params['PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY']
+        endpoint = 0
+        command = Clusters.GeneralDiagnostics.Commands.PayloadTestRequest(
+            enableKey=enable_key,
+            value=ord('A'),
+            count=800
+        )
+        invoke_request_1 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+
+        command = Clusters.OperationalCredentials.Commands.CertificateChainRequest(
+            Clusters.OperationalCredentials.Enums.CertificateChainTypeEnum.kDACCertificate)
+        invoke_request_2 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+
+        try:
+            result = await dev_ctrl.TestOnlySendBatchCommands(dut_node_id, [invoke_request_1, invoke_request_2])
+        except InteractionModelError:
+            asserts.fail("DUT failed to respond to")
+
+        responses = result.Responses
+        asserts.assert_equal(len(responses), 2, "Unexpected number of InvokeResponses sent back from DUT")
+        asserts.assert_true(type_matches(
+            responses[0], Clusters.GeneralDiagnostics.Commands.PayloadTestResponse), "Unexpected return type for first InvokeRequest")
+        asserts.assert_true(type_matches(
+            responses[1], Clusters.OperationalCredentials.Commands.CertificateChainResponse), "Unexpected return type for second InvokeRequest")
+        logging.info("DUT successfully responded to a InvokeRequest action with two valid commands")
+
+        asserts.assert_equal(responses[0].payload, b'A' * 800, "Expect response to match for count == 800")
+        # If this assert fails then some assumptions we were relying on are now no longer true.
+        asserts.assert_greater_equal(result.ResponseMessageCount, 2, "DUT was expected to send multiple InvokeResponseMessages")
```

Comment:
Doesn't this duplicate `asserts.assert_equal(len(responses), 2, "Unexpected number of InvokeResponses sent back from DUT")` above?